File: Developer/AppFrameworks/UIKit/YJCollectionView/YJCollectionView/YJTestCollectionReusableView.m
//
//  YJTestCollectionReusableView.m
//  YJCollectionView
//
//  Created by 阳君 on 16/7/8.
//  Copyright © 2016年 YJ. All rights reserved.
//

#import "YJTestCollectionReusableView.h"

@implementation YJTestCollectionReusableViewModel

@end

@implementation YJTestCollectionReusableView

+ (CGSize)collectionViewManager:(YJUICollectionViewManager *)collectionViewManager viewForSupplementaryElementOfKind:(NSString *)kind referenceSizeForCellObject:(YJUICollectionCellObject *)cellObject {
    if (cellObject.createCell == YJUICollectionCellCreateClass) {
        return CGSizeMake(10, 100);
    }
    return [super collectionViewManager:collectionViewManager viewForSupplementaryElementOfKind:kind referenceSizeForCellObject:cellObject];
}

- (void)reloadDataAsyncWithCellObject:(YJUICollectionCellObject *)cellObject collectionViewManager:(YJUICollectionViewManager *)collectionViewManager {
    [super reloadDataAsyncWithCellObject:cellObject collectionViewManager:collectionViewManager];
    YJTestCollectionReusableViewModel *cm = cellObject.cellModel;
    self.backgroundColor = cm.backgroundColor;
}

@end

File: Demo/OC/AppFrameworks/UIKit/YJCollectionView/YJCollectionView/Cell/YJTestCollectionReusableView.swift
//
//  YJTestCollectionReusableView.swift
//  YJCollectionView
//
//  Created by 阳君 on 2019/7/1.
//  Copyright © 2019 YJ. All rights reserved.
//

import UIKit
import YJCocoa

@objcMembers
class YJTestCollectionReusableViewModel: NSObject {
    /// 背景色
    var backgroundColor: UIColor!
}

@objcMembers
class YJTestCollectionReusableView: YJUICollectionReusableView {
    
    override class func collectionViewManager(_ collectionViewManager: YJUICollectionViewManager, referenceSizeFor kind: String, in cellObject: YJUICollectionCellObject) -> CGSize {
        return CGSize(width: 10, height: 100)
    }
    
    override func collectionViewManager(_ collectionViewManager: YJUICollectionViewManager, reloadWith cellObject: YJUICollectionCellObject) {
        super.collectionViewManager(collectionViewManager, reloadWith: cellObject)
        let cm = cellObject.cellModel as? YJTestCollectionReusableViewModel
        self.backgroundColor = cm?.backgroundColor
    }
    
}
